### PR TITLE
Rename StarshipEngine to RemoteEngine, starship CLI to sf

### PR DESCRIPTION
**Pull request by josh146**
This is the last change to be merged into the demo branch, before demo is merged into master.

### CODE REVIEW

File: doc/introduction/remote.rst [hunk 1]
```diff
@@ -1,25 +1,25 @@
 .. _starship:
```
Comment (josh146) [inline doc/introduction/remote.rst]: The only reminder 😿

File: tests/frontend/test_circuitspecs_chip2.py [hunk 2]
```diff
@@ -368,28 +369,32 @@ def test_no_unitary(self, tol):
...
-            ops.MZgate(0, 0) | (q[2], q[3])
+            # This can be easily seen from below by noting that:
+            # MZ(pi, pi) = R(0) = I
+            # MZ(pi, 0) @ MZ(pi, 0) = I
+            # R(pi) @ MZ(pi, 0) = I
+            ops.MZgate(np.pi, 0) | (q[0], q[1])
+            ops.MZgate(np.pi, 0) | (q[2], q[3])
+            ops.MZgate(np.pi, np.pi) | (q[1], q[2])
+            ops.MZgate(np.pi, np.pi) | (q[0], q[1])
+            ops.MZgate(np.pi, 0) | (q[2], q[3])
```
Comment (josh146) [inline tests/frontend/test_circuitspecs_chip2.py]: Note: the change here was required due to merging in #311. This altered the way the decomposition represents the identity unitary. Have added a comment to help explain why this is the identity.

File: tests/frontend/test_circuitspecs_chip2.py [hunk 3]
```diff
@@ -368,28 +369,32 @@ def test_no_unitary(self, tol):
...
-            ops.MZgate(0, 0) | (q[2], q[3])
+            # This can be easily seen from below by noting that:
+            # MZ(pi, pi) = R(0) = I
+            # MZ(pi, 0) @ MZ(pi, 0) = I
+            # [R(pi) \otimes I] @ MZ(pi, 0) = I
+            ops.MZgate(np.pi, 0) | (q[0], q[1])
+            ops.MZgate(np.pi, 0) | (q[2], q[3])
+            ops.MZgate(np.pi, np.pi) | (q[1], q[2])
+            ops.MZgate(np.pi, np.pi) | (q[0], q[1])
+            ops.MZgate(np.pi, 0) | (q[2], q[3])
```
Comment (josh146) [inline tests/frontend/test_circuitspecs_chip2.py]: Note: the change here was required due to merging in #311. This altered the way the decomposition represents the identity unitary. Have added a comment to help explain why this is the identity.

File: tests/frontend/test_circuitspecs_chip2.py [hunk 4]
```diff
@@ -359,6 +359,7 @@ def test_no_unitary(self, tol):
             ops.MeasureFock() | q
 
         res = prog.compile("chip2")
+        res.print()
```
Comment (antalszava) [inline tests/frontend/test_circuitspecs_chip2.py]: Was this change also required due to #311?

Comment (josh146) [reply tests/frontend/test_circuitspecs_chip2.py]: Nope, this was me debugging, good catch!
```suggestion
```

**Comment by antalszava** [comment]
One comment before merge: updating the `CHANGELOG`. :) As `starship` appeared in there a couple of times, it would explain earlier PR entries in there to the users.

**Comment by josh146** [comment]
> One comment before merge: updating the CHANGELOG. :) As starship appeared in there a couple of times, it would explain earlier PR entries in there to the users.

Good catch. I'm thinking it makes sense to accumulate those entries into one entry, summarising the new RemoteEngine and related functionality, since this will be a single squash commit into master. We did a similar thing when we released the apps layer

**Comment by josh146** [comment]
Note: tests are failing due to an API change in the new release of pytest. I'm going to temporarily pin travis to the previous pytest version, and then the test suite can be upgraded in a subsequent PR (as it will be quite a lot of changes)

**Pull request merged**